# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmSignalHandlers.cpp
# Commit: caa8552f4f52
# Full Hash: caa8552f4f5298399e9e9193a623bc3007652df5
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-18 09:41:05
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r 43921ad138e3 -r caa8552f4f52 js/src/wasm/WasmSignalHandlers.cpp
--- a/js/src/wasm/WasmSignalHandlers.cpp	Mon Jun 08 20:56:09 2020 +0000
+++ b/js/src/wasm/WasmSignalHandlers.cpp	Thu Jun 18 05:33:30 2020 +0000
@@ -707,7 +707,7 @@
   // due to this trap occurring in the indirect call prologue, while fp points
   // to the caller's Frame which can be in a different Module. In any case,
   // though, the containing JSContext is the same.
-  Instance* instance = ((Frame*)ContextToFP(context))->tls->instance;
+  Instance* instance = ((Frame*)ContextToFP(context))->instance();
   MOZ_RELEASE_ASSERT(&instance->code() == &segment.code() ||
                      trap == Trap::IndirectCallBadSig);
 
@@ -1167,7 +1167,7 @@
     return false;
   }
 
-  Instance& instance = *reinterpret_cast<Frame*>(regs.fp)->tls->instance;
+  Instance& instance = *Frame::fromUntaggedWasmExitFP(regs.fp)->instance();
   MOZ_ASSERT(&instance.code() == &segment.code());
 
   if (!instance.memoryAccessInGuardRegion((uint8_t*)addr, numBytes)) {